splash_ingest/server/poller.py

```diff
@@ -48,10 +52,31 @@ def init_logging():
 ingest_db = MongoClient(INGEST_DB_URI)[INGEST_DB_NAME]
 
 init_ingest_service(ingest_db, databroker_db)
+
+
+class TerminateRequested():
+    state = False
+
+
+terminate_requested = TerminateRequested
+
 poll_for_new_jobs(
     POLLER_SLEEP_SECONDS,
     SCICAT_BASEURL,
     SCICAT_INGEST_USER,
     SCICAT_INGEST_PASSWORD,
+    terminate_requested,
     THUMBS_ROOT
 )
+
+
+def sigterm_handler(signum, frame):
+    logger.info("sigterm received")
+    terminate_requested.state = True
```

Comment:
Put the original handler back now?

Reply:
Not sure I understand

Reply:
Related to other comment. Forgetting the content manager, would you stash the original handler globally and then reinstate it here once your handler has fired. Maybe more than you need here though...